pkg/dataset/downsample.go

```diff
@@ -0,0 +1,76 @@
+// This file and its contents are licensed under the Apache License 2.0.
+// Please see the included NOTICE for copyright information and
+// LICENSE for a copy of the license.
+
+package dataset
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/jackc/pgx/v4"
+
+	"github.com/timescale/promscale/pkg/log"
+	"github.com/timescale/promscale/pkg/rollup"
+)
+
+const (
+	defaultDownsampleState      = true
+	defaultDownsampleResolution = "short:5m:90d,long:1h:395d" // label:resolution:retention
+)
+
+var (
+	setDefaultDownsampleStateSQL      = "SELECT prom_api.set_automatic_downsample($1)"
+	setDefaultDownsampleResolutionSQL = "SELECT prom_api.set_downsample_resolution($1)"
+
+	defaultDownsampleStateVar = defaultDownsampleState
+)
+
+type Downsample struct {
+	Automatic  *bool  `yaml:"automatic,omitempty"`
+	Resolution string `yaml:"resolution,omitempty"`
```

Comment:
I think having a list of resolutions would be easier to navigate. Strings are tedious to parse/understand and more error prone.
```
type Resolution struct {
   Label string
   Interval time.Duration
   Retention time.Duration
}
```